### ISSUE TITLE
multiple lock message prompts

### ISSUE THREAD

**Issue by nsturnitin** [maruf-freelancer/capistrano-deploy-lock#2]
*\* Invoke deploy:lock (first_time)
*\* Execute deploy:lock
**Please enter lock message (): Please enter lock message (): Please enter lock message ():**

fixed by taking logic out of roles loop:

desc "Set deploy lock with a custom lock message and expiry time"
  task :lock do

```
set :lock_message, ask('lock message', '', echo: true)
puts "Lock message: #{fetch(:lock_message)}"
set :custom_deploy_lock, true
while fetch(:lock_expiry).nil?
  set :expire_after, ask('minutes of expiry', 'optional', echo: true)
  expire_after = fetch(:expire_after)

  if expire_after == "optional"
    # Never expire an explicit lock if no time given
    set :lock_expiry, false
    puts colorize("Lock will never expire automatically.", color: 33)
  else
    unless expire_after.to_i == 0
      set :lock_expiry, (Time.now + expire_after.to_i * 60).utc
      puts colorize("Expire after #{fetch(:expire_after)} minutes", color: 33)
    else
      puts colorize("'#{expire_after}' is not valid input. Please try again.")
    end
  end
end

on roles(fetch(:deploy_lock_roles)) do
  with rails_env: fetch(:deploy_lock_rack_env) do
    invoke 'deploy:create_lock'
  end
end
```

  end

**Comment by nsturnitin** [comment]
This is a symptom of using the on roles that invokes a task that also has on roles (nested onroles).  I can only guess that the environment that this was written for only had one deployment machine with 2 different target servers so this was not seen in your testing.

I think this was specific to your environment and should be removed:
 with rails_env: fetch(:deploy_lock_rack_env) do

overall, pretty handy code, but not as flexible as I needed for our environment.

**Comment by mhbcse** [comment]
Thanks for your valuable input. Solved the problem in version 1.0.3
